Add make release to Staging branch

aiyengar2 opened the pull request:
Add the ability to quickly move assets into the released/ directory and update the index.yamls.

On release, a release captain is expected to run this script to get the Staging branch ready to absorb commits from a new release.

Devs still will be forced to update the packageVersion since the Source branch validates against already released assets in the Live branch anyways.

aiyengar2 commented:
Once this has been reviewed, this needs to be backported to `rancher/charts-build-scripts` templates to allow other repositories to pull it in / use it as well.

cbron commented on scripts/release-assets:
```diff
@@ -0,0 +1,20 @@
+rm -R $(ls -1 -d charts/*/) $(ls -1 -d assets/*/)
+
+helm repo index --merge ./released/assets/index.yaml --url released/assets released/assets
+cp released/assets/index.yaml index.yaml
```
> We should also add a README.md to the released/ directory to allow us to officially document what this directory is meant for.

Yea clarifying this somewhere would be good.

aiyengar2 replied:
will add this.

cbron commented on scripts/release-assets:
```diff
@@ -0,0 +1,20 @@
+fi
+
+cp -R assets/ released/assets
+rm released/assets/README.md
```
why no readme ?

aiyengar2 replied:
The README.md is meant for the assets/ directory, the released/assets/ directory should probably have its own README.md declared in the released/ directory instead. I'll add that README.md in to explain the purpose of this folder to the charts-build-scripts PR.

aiyengar2 commented:
Closing this PR out in favor of tracking this all in https://github.com/rancher/charts-build-scripts/pull/15 instead.

@cbron I should have addressed your comments in the other PR. README.md has been added for the `released/` directory.
